### DISCUSSION TITLE
After upgrading to 16.9.1 p-listbox onClick argument is the previous value

### DISCUSSION THREAD

**Discussion by aabot99** [primefaces/community#656] [PrimeNG] [answered]
I upgraded my primeng from 16.0.2 to 16.9.1 and encountered an issue with the p-listbox.  What happens is when the user clicks on the next entry, the previous entry is passed on the onClick handler.  This is not happening in the 16.0.2 version.  I reviewed the API and could not figure out why.

Here is the HTML snippet that is shared between both versions:
![image](https://github.com/primefaces/community/assets/152433894/fc73bfe6-ae89-4758-9d0a-4b3aa760248a)

Here is the handler code in debugging mode showing the value:
![image](https://github.com/primefaces/community/assets/152433894/57ebe061-5bc0-4a6e-a736-7154b8f178dc)

And lastly here is the listbox showing the current entry highlighted (Group Selection) even though the click handler's argument was for the select:
![image](https://github.com/primefaces/community/assets/152433894/c532c4c9-6522-41bf-b1b3-bae68ba8bca1)

UPDATE: After trying a few different approaches, I discovered that onChange event contains the selected value,  Is this going to be the behavior from here on or are you going to switch back where the onClick event will contain what was just clicked on?

Thanks, -A

**Comment by cetincakiroglu** [comment]
Could you please share a stackblitz reproducer so we can identify the issue?

**Reply by aabot99** [reply, marked as answer]
After upgrading to 17.3.1 I just switched to that handler.  It's working now,  Thanks.
